### PR TITLE
tests: make tests run independently

### PR DESCRIPTION
**Pull request by mareklabonarski**

### CODE REVIEW

File: tests/it/it_scargo_commands_flow.py [hunk 1]
```diff
@@ -466,8 +718,9 @@ def test_cli_clean_after_build_profile_release(
             "build"
         ).is_dir(), "Dictionary 'build' still exist when 'clean' should remove it"
 
-    @pytest.mark.order(after="test_cli_check")
-    def test_cli_test(self, test_state: ActiveTestState) -> None:
+    def test_cli_test(
+        self, test_state: ActiveTestState, copy_project_and_update2: Any
+    ) -> None:  # TODO for sure?
```
[the remark this answers is not in the document]
Comment (aaksenczuk) [reply tests/it/it_scargo_commands_flow.py]: fixed

File: tests/it/it_scargo_commands_flow.py [hunk 2]
```diff
@@ -176,39 +176,295 @@ def get_expected_files_list_new_proj_bin(src_dir_name: str, bin_name: str) -> Li
...
+        target_id=TargetIds.stm32,
+        proj_name=TEST_PROJECT_STM32_NAME,
+        proj_to_copy_path=TEST_PROJECT_STM32_PATH,
+    )
+
+    return state
+
+
+@pytest.fixture
+def test_state(request: Any, tmpdir_factory: Any) -> Any:
```
[the remark this answers is not in the document]
Comment (aaksenczuk) [reply tests/it/it_scargo_commands_flow.py]: fixed

File: tests/it/it_scargo_commands_flow.py [hunk 3]
```diff
@@ -176,39 +176,295 @@ def get_expected_files_list_new_proj_bin(src_dir_name: str, bin_name: str) -> Li
...
+    assert (
+        project_path is not None
+    ), f"Project not copied under expected location: {project_path}"
+    docker_path = Path(project_path, ".devcontainer")
+    config = parse_config(project_path / "scargo.lock")
+    generate_env(docker_path, config)
+
+
+@pytest.fixture
+def copy_project_and_update2(test_state: ActiveTestState, copy_project: Any) -> None:
```
[the remark this answers is not in the document]
Comment (aaksenczuk) [reply tests/it/it_scargo_commands_flow.py:279]: copy_project_and_update and other return None, I think then shall have Any on input

File: tests/it/it_scargo_commands_flow.py [hunk 4]
```diff
@@ -176,39 +176,295 @@ def get_expected_files_list_new_proj_bin(src_dir_name: str, bin_name: str) -> Li
...
+def copy_project(test_state: ActiveTestState) -> None:
+    if test_state.proj_to_copy_path is None:
+        pytest.skip("Test of new project - no copying needed")
+    os.makedirs(f"{test_state.proj_path}/{test_state.proj_name}", exist_ok=True)
+    os.chdir(f"{test_state.proj_path}/{test_state.proj_name}")
+    copytree(test_state.proj_to_copy_path, os.getcwd(), dirs_exist_ok=True)
+
+
+@pytest.fixture
+def copy_project_and_update(test_state: ActiveTestState, copy_project: Any) -> None:
```
[the remark this answers is not in the document]
Comment (mareklabonarski) [reply tests/it/it_scargo_commands_flow.py:268]: fixtures are just functions in our case, so `callable` is a good type for them,

File: tests/it/it_scargo_commands_flow.py [hunk 5]
```diff
@@ -176,39 +176,295 @@ def get_expected_files_list_new_proj_bin(src_dir_name: str, bin_name: str) -> Li
...
+    os.chdir(f"{test_state.proj_path}/{test_state.proj_name}")
+    result = test_state.runner.invoke(cli, ["fix"])
+    assert (
+        result.exit_code == 0
+    ), f"Command 'fix' end with non zero exit code: {result.exit_code}"
+
+
+@pytest.fixture
+def new_profile_project_build_fix_check(
+    test_state: ActiveTestState, new_profile_project_build: Any
```
[the remark this answers is not in the document]
Comment (aaksenczuk) [reply tests/it/it_scargo_commands_flow.py]: fixed

File: tests/it/it_scargo_commands_flow.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aaksenczuk) [reply tests/it/it_scargo_commands_flow.py]: fixed

File: tests/it/it_scargo_commands_flow.py [hunk 6]
```diff
@@ -9,12 +9,12 @@
 from enum import Enum
 from pathlib import Path
 from shutil import copytree
-from typing import List, Optional
+from typing import Any, List, Optional
```
[the remark this answers is not in the document]
Comment (aaksenczuk) [reply tests/it/it_scargo_commands_flow.py:12]:
```
@pytest.fixture
def test_state(request: FixtureRequest, tmpdir_factory: TempdirFactory) -> Any:
    test_state = request.param
    test_state.proj_path = tmpdir_factory.mktemp(test_state.proj_name)
    return test_state
```    
    we need it here

**Pull request merged**